Inconsistent base type and self type using Linear Cell Complex with cell attribute with point and id, when Info_!=void

## Issue Details

The typedef `Self `and `Base1` in the class Cell_attribute_with_point are wrongly declared.
Using a Linear Cell Complex with cell attribute with point and id, when Info_!=void:


```
Cell_attribute_with_point.h:108:18: error: type ‘CGAL::Cell_attribute_with_point<CGAL::CMap_linear_cell_complex_storage_with_index<2, 2, CGAL::Linear_cell_complex_traits<2, CGAL::Epick>, dartItem, std::allocator<int> >, std::vector<double>, CGAL::Boolean_tag<true>, CGAL::Null_functor, CGAL::Null_functor, CGAL::Boolean_tag<true> >::Base1’ {aka ‘CGAL::Cell_attribute<CGAL::CMap_linear_cell_complex_storage_with_index<2, 2, CGAL::Linear_cell_complex_traits<2, CGAL::Epick>, dartItem, std::allocator<int> >, std::vector<double>, CGAL::Boolean_tag<true>, CGAL::Null_functor, CGAL::Null_functor, CGAL::Boolean_tag<false> >’} is not a base type for type ‘CGAL::Cell_attribute_with_point<CGAL::CMap_linear_cell_complex_storage_with_index<2, 2, CGAL::Linear_cell_complex_traits<2, CGAL::Epick>, dartItem, std::allocator<int> >, std::vector<double>, CGAL::Boolean_tag<true>, CGAL::Null_functor, CGAL::Null_functor, CGAL::Boolean_tag<true> >’
  108 |     using Base1::info;
      |                  ^~~~
```

## Source Code

```c++
struct dartItem
{
    using Use_index=CGAL::Tag_true; // use indices
    using Index_type=std::size_t;
    template<class Refs>
    struct Dart_wrapper
    {
        using Vertex_attribute = CGAL::Cell_attribute_with_point_and_id<Refs, std::vector<double>, CGAL::Tag_true>;
        using Edge_attribute   = CGAL::Cell_attribute_with_point_and_id<Refs, std::vector<double>, CGAL::Tag_true>;
        using Facet_attribute  = CGAL::Cell_attribute_with_point_and_id<Refs, std::vector<double>, CGAL::Tag_true>;
        using Attributes       = std::tuple<Vertex_attribute, Edge_attribute, Facet_attribute>;
    };
};

using Kernel            = CGAL::Exact_predicates_inexact_constructions_kernel;
using cgalLCCTraits     = CGAL::Linear_cell_complex_traits<2, Kernel> ;
using LCC_2             = CGAL::Linear_cell_complex_for_generalized_map<2, 2, cgalLCCTraits, dartItem>;

int main(){
LCC_2 t_lcc2;
}
```

## Solution
Change the typedef 
```c++
    typedef Cell_attribute_with_point<LCC, Info_, Tag, Functor_on_merge_,
                                      Functor_on_split_> Self;

    typedef Cell_attribute<LCC, Info_, Tag,
                           Functor_on_merge_, Functor_on_split_> Base1;
```
to

```c++
    typedef Cell_attribute_with_point<LCC, Info_, Tag, Functor_on_merge_,
                                      Functor_on_split_, WithID> Self;

    typedef Cell_attribute<LCC, Info_, Tag,
                           Functor_on_merge_, Functor_on_split_, WithID> Base1;
```

--- a/Combinatorial_map/include/CGAL/Combinatorial_map_storages_with_index.h
+++ b/Combinatorial_map/include/CGAL/Combinatorial_map_storages_with_index.h
@@ -75,7 +75,7 @@ namespace CGAL {
     struct Container_for_attributes : public
         Compact_container_with_index<T,
         typename Allocator_traits::template rebind_alloc<T>,
-        Multiply_by_two_policy_for_cc_with_size<64>, size_type >
+        Multiply_by_two_policy_for_cc_with_size<64>, Index_type >
     {};
     /// Typedef for attributes
     typedef typename internal::template Get_attributes_tuple<Dart_wrapper>::type

--- a/Generalized_map/include/CGAL/Generalized_map_storages_with_index.h
+++ b/Generalized_map/include/CGAL/Generalized_map_storages_with_index.h
@@ -75,7 +75,7 @@ namespace CGAL {
     struct Container_for_attributes : public
         Compact_container_with_index<T,
         typename Allocator_traits::template rebind_alloc<T>,
-        Multiply_by_two_policy_for_cc_with_size<64>, size_type >
+        Multiply_by_two_policy_for_cc_with_size<64>, Index_type >
     {};
     /// Typedef for attributes
     typedef typename internal::template Get_attributes_tuple<Dart_wrapper>::type

--- a/Linear_cell_complex/include/CGAL/CMap_linear_cell_complex_storages_with_index.h
+++ b/Linear_cell_complex/include/CGAL/CMap_linear_cell_complex_storages_with_index.h
@@ -85,7 +85,7 @@ namespace CGAL {
     struct Container_for_attributes : public
         Compact_container_with_index<T,
         typename Allocator_traits::template rebind_alloc<T>,
-        Multiply_by_two_policy_for_cc_with_size<64>, size_type >
+        Multiply_by_two_policy_for_cc_with_size<64>, Index_type >
     {};
     /// Typedef for attributes
     typedef typename internal::template Get_attributes_tuple<Dart_wrapper>::type

--- a/Linear_cell_complex/include/CGAL/Cell_attribute_with_point.h
+++ b/Linear_cell_complex/include/CGAL/Cell_attribute_with_point.h
@@ -91,10 +91,10 @@ namespace CGAL {
 
   public:
     typedef Cell_attribute_with_point<LCC, Info_, Tag, Functor_on_merge_,
-                                      Functor_on_split_> Self;
+                                      Functor_on_split_, WithID> Self;
 
     typedef Cell_attribute<LCC, Info_, Tag,
-                           Functor_on_merge_, Functor_on_split_> Base1;
+                           Functor_on_merge_, Functor_on_split_, WithID> Base1;
     typedef Point_for_cell<typename LCC::Point> Base2;
 
     typedef typename LCC::Point             Point;

--- a/Linear_cell_complex/include/CGAL/Cell_attribute_with_point_and_id.h
+++ b/Linear_cell_complex/include/CGAL/Cell_attribute_with_point_and_id.h
@@ -32,6 +32,9 @@ namespace CGAL {
     template <class, class>
     friend class Concurrent_compact_container;
 
+    template <class, class, class, class>
+    friend class Compact_container_with_index;
+
   public:
     typedef typename Base::Point Point;
 
@@ -63,6 +66,9 @@ namespace CGAL {
 
     template <class, class>
     friend class Concurrent_compact_container;
+
+    template <class, class, class, class>
+    friend class Compact_container_with_index;
 
   public:
     typedef typename Base::Point Point;

--- a/Linear_cell_complex/include/CGAL/GMap_linear_cell_complex_storages_with_index.h
+++ b/Linear_cell_complex/include/CGAL/GMap_linear_cell_complex_storages_with_index.h
@@ -85,7 +85,7 @@ namespace CGAL {
     struct Container_for_attributes : public
         Compact_container_with_index<T,
         typename Allocator_traits::template rebind_alloc<T>,
-        Multiply_by_two_policy_for_cc_with_size<64>, size_type >
+        Multiply_by_two_policy_for_cc_with_size<64>, Index_type >
     {};
     /// Typedef for attributes
     typedef typename internal::template Get_attributes_tuple<Dart_wrapper>::type

--- a/Linear_cell_complex/test/Linear_cell_complex/Linear_cell_complex_2_test.cpp
+++ b/Linear_cell_complex/test/Linear_cell_complex/Linear_cell_complex_2_test.cpp
@@ -54,6 +54,7 @@ struct Myitems_2b
 {
 #ifdef USE_COMPACT_CONTAINER_WITH_INDEX
   typedef CGAL::Tag_true Use_index;
+  using Index_type=std::uint16_t;
 #endif
   template <class LCC>
   struct Dart_wrapper

--- a/Linear_cell_complex/test/Linear_cell_complex/Linear_cell_complex_3_test.cpp
+++ b/Linear_cell_complex/test/Linear_cell_complex/Linear_cell_complex_3_test.cpp
@@ -60,6 +60,7 @@ struct Myitems_3b
 {
 #ifdef USE_COMPACT_CONTAINER_WITH_INDEX
   typedef CGAL::Tag_true Use_index;
+  using Index_type=std::uint32_t;
 #endif
   template <class LCC>
   struct Dart_wrapper

--- a/Linear_cell_complex/test/Linear_cell_complex/Linear_cell_complex_4_test.cpp
+++ b/Linear_cell_complex/test/Linear_cell_complex/Linear_cell_complex_4_test.cpp
@@ -57,6 +57,7 @@ struct Myitems_4b
 {
 #ifdef USE_COMPACT_CONTAINER_WITH_INDEX
   typedef CGAL::Tag_true Use_index;
+  using Index_type=std::uint64_t;
 #endif
   template <class LCC>
   struct Dart_wrapper

